### PR TITLE
Import/Export site customizations

### PR DESCRIPTION
**Pull request by riking**
Finally, we can have an easy way to share customizations!

image

image

### CODE REVIEW

**Comment by riking** [comment]
Test failures are unrelated.

File: app/assets/javascripts/discourse/components/json-file-uploader.js.es6 [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    const self = this;
+
+    const $fileInput = $this.find('#js-file-input');
+    this.set('fileInput', $fileInput[0]);
+
+    $fileInput.on('change', function() {
+      self.fileSelected(this.files);
+    });
+
+    const $dragContainer = $this.find('.jsfu-shade-container');
```
Comment (SamSaffron) [inline app/assets/javascripts/discourse/components/json-file-uploader.js.es6]: supporting drag and drop here seems like a massive amount of gold plating. not sure we need it .

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -143,6 +143,7 @@ en:
     every_two_weeks: "every two weeks"
     every_three_days: "every three days"
     max_of_count: "max of {{count}}"
+    alternation: "or"
```
Comment (SamSaffron) [inline config/locales/client.en.yml]: "or" will never translate right, you need to translate the whole sentence

Comment (riking) [inline config/locales/client.en.yml]: Yeah, wasn't sure about the design here.

Any suggestions? Here is where it's used:

File: app/assets/javascripts/discourse/templates/modal/upload-customization.hbs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<form {{action "dummy" on="submit"}}>
+  <div class='modal-body'>
+    {{json-file-uploader value=customizationFile extension=".dcstylejson"}}
```
Comment (SamSaffron) [inline app/assets/javascripts/discourse/templates/modal/upload-customization.hbs]: really not a fan of dcstylejson its a real weird extension.

either 

dcstyle

or 

dcstyle.json if we must

File: app/assets/javascripts/discourse/components/json-file-uploader.js.es6 (hunk 1 above)
Comment (riking) [inline app/assets/javascripts/discourse/components/json-file-uploader.js.es6]: I tried using the system and found that drag&drop was needed for a nice flow, at least in Chrome.

 1. go to meta, click download button on theme, it goes to bottom of window
 2. go to your site, click import
 3. drag file from the bottom up into the import window

File: app/assets/javascripts/discourse/templates/modal/upload-customization.hbs (hunk 3 above)
Comment (riking) [inline app/assets/javascripts/discourse/templates/modal/upload-customization.hbs]: Agreed, will revert to `.dcstyle.json`

**Comment by coding-horror** [comment]
we really should pull this in, maybe as soon as 1.4 is ongoing?

**Pull request merged**